### ISSUE TITLE
🐛 Bug Report: a file cannot be deleted, and yields a 404 when calling getFileDownload

### ISSUE THREAD

**Issue by t0mm4rx** [appwrite/appwrite#5965]
### 👟 Reproduction steps

Unfortunately, this error is quite random and I had it only once in 100 files.

1. Create a large file (> 30MB)
2. Try to delete it
3. It yields a 500 error:
```json
{
    "message": "Server Error",
    "code": 500,
    "type": "general_server_error",
    "version": "1.3.7"
}
```
And in the `appwrite` container:
```
appwrite  | [Error] Method: DELETE
appwrite  | [Error] URL: /v1/storage/buckets/:bucketId/files/:fileId
appwrite  | [Error] Type: Appwrite\Extend\Exception
appwrite  | [Error] Message: Failed to delete file from device
appwrite  | [Error] File: /usr/src/code/app/controllers/api/storage.php
appwrite  | [Error] Line: 1436
```
4. Now it's impossible to get the file, it returns a 404. It's also impossible to delete the file.

The error that is triggered in the `appwrite` container while getting the file:
```
appwrite  | [Error] Timestamp: 2023-08-09T10:47:08+00:00
appwrite  | [Error] Method: GET
appwrite  | [Error] URL: /v1/storage/buckets/:bucketId/files/:fileId/download
appwrite  | [Error] Type: Appwrite\Extend\Exception
appwrite  | [Error] Message: File not found in /storage/uploads/app-645e685e270db5ef8924/csv/64d0f33923dc1e49a3a3.csv
appwrite  | [Error] File: /usr/src/code/app/controllers/api/storage.php
appwrite  | [Error] Line: 1008
appwrite  | [Error] Timestamp: 2023-08-09T10:47:15+00:00
```

### 👍 Expected behavior

We should be able to delete files.
We should be able to get files.

### 👎 Actual Behavior

Sometimes, file cannot be deleted.

### 🎲 Appwrite version

Version 1.3.x

### 💻 Operating system

Linux

### 🧱 Your Environment

Self-hosted

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/.github/blob/main/CODE_OF_CONDUCT.md)

**Comment by t0mm4rx** [comment]
It appears that the file has been deleted from the file system, but not in the database. Maybe you could implement some kind of sanity check for this scenario.

**Comment by safwanyp** [comment]
Alright so I've been doing some reproduction tests. I was able to upload around 20 60MB files, and they seemed to delete with no issue. 

I also checked the mariadb entries, which showed 0 files, meaning deletion was successful. 

Any specific time this happens? Or it's completely random?

**Comment by t0mm4rx** [comment]
It seems completely random.
This is in a production environment so while the file was deleted some read / update / delete operations were taking place in parallel on databases and other files.

Maybe you can replicate the issue directly by deleting the file in the `storage` container, and try interacting with the file.

**Comment by safwanyp** [comment]
@t0mm4rx Just to be clear, the read/update/delete operations you mentioned were being performed through your own app and not by Appwrite itself, right?

I'll try to reproduce this once again!

**Comment by t0mm4rx** [comment]
Yes, these operations were performed by the users of the app, interacting with Appwrite through the various SDKs.

**Comment by Haimantika** [comment]
Hi @t0mm4rx thanks for raising this issue. I will share it with my team and get back with possible solutions/why this is happening.

**Comment by abnegate** [comment]
@t0mm4rx I'm fairly certain this was related to a permissions issue fixed in 1.4.0, as such I'm going to close the issue. Please re-open if this is still occuring for you :+1:

**Comment by t0mm4rx** [comment]
Hello, this is still an issue in 1.4.12.

Appwrite logs:
```
Warning: scandir(/storage/uploads/app-645e685e270db5ef8924/csv/tmp_ugg.csv.csv/): Failed to open directory: No such file or directory in /usr/src/code/vendor/utopia-php/storage/src/Storage/Device/Local.php on line 526

Warning: scandir(): (errno 2): No such file or directory in /usr/src/code/vendor/utopia-php/storage/src/Storage/Device/Local.php on line 526

Warning: foreach() argument must be of type array|object, bool given in /usr/src/code/vendor/utopia-php/storage/src/Storage/Device/Local.php on line 526

Warning: rmdir(/storage/uploads/app-645e685e270db5ef8924/csv/tmp_ugg.csv.csv/): No such file or directory in /usr/src/code/vendor/utopia-php/storage/src/Storage/Device/Local.php on line 252
[Error] Timestamp: 2024-02-28T10:44:03+00:00
[Error] Method: DELETE
[Error] URL: /v1/storage/buckets/:bucketId/files/:fileId
[Error] Type: Appwrite\Extend\Exception
[Error] Message: Failed to delete file from device
[Error] File: /usr/src/code/app/controllers/api/storage.php
[Error] Line: 1475
```

My app is now stuck because when reading all files in a bucket it yields a 500. I can't delete the file (500 as well).
I'll try to re-create a new bucket.
